packages/widget/src/components/transfer/fungible/fungible-token-transfer.ts

```diff
@@ -85,6 +88,14 @@ export class FungibleTokenTransfer extends BaseComponent {
     )!;
   }
 
+  protected updated(changedProperties: PropertyValues): void {
+    if (changedProperties.has('network')) {
+      void this.walletController.switchChain(this.network?.chainId as number);
```

Comment:
- please move this logic into `onNetworkSelected` handler of source network selector (L124)
- You should switch network only if network is not undefined